honeycombio/honeycomb-network-agent
# maint: adding proper smoke test(s)

## Which problem is this PR solving?

- Closes #78 

## Description of the changes

Set up a smoke test, details included in the new make targets below:
- `smokey_cluster_create`: Create kind cluster
- `smokey_collector_install`: Install collector via helm with new collector values, including a filter processor that only keeps echoserver spans from pods with a response code of 405
- `smokey_agent_install`: Install agent via helm with new agent values, including usage of a local image (installed if not available in the shell) and export to the collector
- `smokey_echo_job`: Install echoserver and run a job that sends a POST request expecting a 405 response
- `smokey_copy_output`: After a short wait, copy output from the newly created file saved with telemetry from collector
- `smokey_verify_output`: Make assertions via the new `verify.bats` bats test (similar to previous tests on other distros and in go auto-instrumentation) that the output contains the expected span and attributes
- `smoke`: Do all of the things above
- `unsmoke`: Tear it all down when done

Also updated PR, main, and release workflows to run the new smoke test.

## How to verify that this has the expected result

- Run `make smoke` and get a bunch of passing tests - also check out the (git ignored) json file that contains the remaining span that's being tested. `make unsmoke` to spin it down.
